### ISSUE TITLE
make .dep doesn't exit successfully

### ISSUE THREAD

**Issue by itrich** [SovereignCloudStack/website#114]
When trying to build the website on a freshly cloned repository, `make .dep` exits  with Error 1:

```
Makefile:17: .dep: No such file or directory
./scripts/collectdeps.py build/index.html.de build/index.html.en build/.htaccess build/robots.txt build/sitemap.xml build/blog/20200915-garloff-ovh.html.de build/blog/20200915-garloff-ovh.html.en build/test.html.de build/test.html.en >.dep
ERROR: tmp/logos.html not readable 
Makefile:15: recipe for target '.dep' failed
make: *** [.dep] Error 1 
```

**Comment by garloff** [comment]
This one is non-trivial, I'm afraid. You need two make to reliably build the web pages.
We could commit the .dep file into the repo to avoid it. I personally dislike keeping generated content in git, but maybe I need to overcome this ...

**Comment by itrich** [comment]
This is what I'm doing so far within my GitHub Action pipline: https://github.com/itrich/website/actions/runs/1037602004
```
   - uses: actions/checkout@v2
    - name: Setup Node.js environment
      uses: actions/setup-node@v2.2.0  
    - name: Install dependencies
      run: |
        sudo apt-get install sed python-markdown imagemagick cleancss
        sudo npm install html-minifier clean-css-cli -g
    - name: make .dep
      continue-on-error: true
      run: make .dep
    - name: make html
      run: make html
    - name: rsync build
      uses: burnett01/rsync-deployments@4.1
```
However, I'm still struggling to avoid the step `make .dep` to fail unsuccessfully. Have you any clues?

**Comment by garloff** [comment]
You could do a `touch -d "2000-01-01" .dep && make .dep` to avoid the exit code killing your pipeline.

**Comment by itrich** [comment]
Seems that the root cause was `tmp/logos.html` not being present before executing `make .dep`, which seems odd.
However, my GitHub Action finally runs flawlessly, see https://github.com/itrich/website/actions 👍

**Comment by garloff** [comment]
The processing is 
`source` --- (md2html/genlogos.py/include.py) ---> `tmp` ----- (minify) ------> `build`.
The real problem is that we source the `.dep` file from the Makefile which itself knows how to generate `.dep`.
We probably need to split the Makefile into two Makefiles to make things clean.
The first Makefile generating all that's needed for `.dep` and then generating `.dep`. And a second that can then cleanly source the `.dep` file.

**Comment by itrich** [comment]
Closed by #124
